packages/core/admin/server/src/services/encryption.ts

```diff
@@ -0,0 +1,57 @@
+import crypto from 'crypto';
+
+const IV_LENGTH = 16; // 16 bytes for AES-GCM IV
+/**
+ * Encrypts a value string using AES-256-GCM.
+ * Returns a string containing IV, encrypted content, and auth tag (all hex-encoded).
+ */
+const encrypt = (value: string) => {
+  const encryptionKey: string = strapi.config.get('admin.secrets.encryptionKey');
+  if (!encryptionKey) {
+    strapi.log.warn('Encryption key is missing from config');
+    return null;
+  }
+  const iv = crypto.randomBytes(IV_LENGTH);
+  const cipher = crypto.createCipheriv('aes-256-gcm', Buffer.from(encryptionKey, 'hex'), iv);
```

Comment:
I don't know if it's only on my end, but I get type errors here and down below

Reply:
no type errors on my side or the linter, anyways this section did change a bit with a new thing I added